backend/app/api/uploads.py

```
@@ -0,0 +1,100 @@
+        raise HTTPException(status_code=400, detail="Path traversal attempt detected.")
+
+    # 4. Offload Blocking File Write to Thread Pool
+    try:
+        await run_in_threadpool(save_file_sync, file.file, file_path)
+    except SizeLimitExceeded:
+        if file_path.exists():
+            file_path.unlink()
+        raise HTTPException(
+            status_code=413,
+            detail="File size exceeds the maximum limit of 100 MB.",
+        )
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)